video2dataset/subsamplers/optical_flow_subsampler.py

```diff
@@ -4,6 +4,92 @@
 import cv2
 import numpy as np
 
+try:
+    from raft import RAFT
+    from raft.utils.utils import InputPadder
+    import torch
+except: # RAFT not installed
+    pass
+
+
+def resize_image_with_aspect_ratio(image, target_shortest_side=16):
+    """
+    Resize an input image while maintaining its aspect ratio.
+
+    This function takes an image and resizes it so that its shortest side
+    matches the specified target length. The other side is scaled accordingly
+    to maintain the original aspect ratio of the image. The function returns
+    the resized image and the scaling factor used for resizing.
+
+    Parameters
+    ----------
+    image : numpy.ndarray
+        The input image represented as a NumPy array with shape (height, width, channels).
+    target_shortest_side : int, optional
+        The desired length for the shortest side of the resized image (default is 16).
+
+    Returns
+    -------
+    resized_image : numpy.ndarray
+        The resized image with the same number of channels as the input image.
+    scaling_factor : float
+        The scaling factor used to resize the image.
+    """
+    # Get the original dimensions of the image
+    height, width = image.shape[:2]
+
+    # Calculate the new dimensions while maintaining the aspect ratio
+    if height < width:
+        new_height = target_shortest_side
+        new_width = int(width * (target_shortest_side / height))
+        scaling_factor = height / new_height
+    else:
+        new_width = target_shortest_side
+        new_height = int(height * (target_shortest_side / width))
+        scaling_factor = width / new_width
+    # Resize the image using the calculated dimensions
+    resized_image = cv2.resize(image, (new_width, new_height), interpolation=cv2.INTER_AREA)
+
+    return resized_image, scaling_factor
+
+class RAFTDetector:
+    def __init__(
+        self, args, downsample_size=None, device='cuda'
+    ):
+        model = torch.nn.DataParallel(RAFT(args))
+        model.load_state_dict(torch.load(args.model))
+
+        model = model.module
+        model.to(device)
+        model.eval()
+
+        self.model = model
+        self.device = device
+        self.downsample_size = downsample_size
+
+    def preprocess(self, frame1, frame2):
```

Comment:
you already said results are good I think but always double triple check if prepro is correct